#
# 1) Stops scrollback files growing too large by fixing the file-shrink code.
# 2) Puts a "Display scrollback from previous session" into the Setup GUI
#    (logging section) so people can turn this off without typing commands.
#
--- xchat-2.8.4/src/common/text.c	29 Jul 2007 05:02:36 -0000	1.90
+++ xchat-2.8.4p1/src/common/text.c	26 Sep 2007 07:29:55 -0000
@@ -285,6 +285,8 @@
 		}
 	}
 
+	sess->scrollwritten = lines;
+
 	if (lines)
 	{
 		text = ctime (&stamp);
--- xchat-2.8.4/src/fe-gtk/setup.c	9 Sep 2007 04:57:07 -0000	1.100
+++ xchat-2.8.4p1/src/fe-gtk/setup.c	26 Sep 2007 07:29:57 -0000
@@ -382,7 +382,8 @@
 static const setting logging_settings[] =
 {
 	{ST_HEADER,	N_("Logging"),0,0,0},
-	{ST_TOGGLE,	N_("Enable logging of conversations"), P_OFFINTNL(logging), 0, 0, 2},
+	{ST_TOGGLE,	N_("Display scrollback from previous session"), P_OFFINTNL(text_replay), 0, 0, 0},
+	{ST_TOGGLE,	N_("Enable logging of conversations"), P_OFFINTNL(logging), 0, 0, 2},
 	{ST_ENTRY,	N_("Log filename:"), P_OFFSETNL(logmask), 0, 0, sizeof prefs.logmask},
 	{ST_LABEL,	N_("%s=Server %c=Channel %n=Network.")},
 
